### ISSUE TITLE
MissingMethodException: RecyclableMemoryStreamManager.GetStream()

### ISSUE THREAD

**Issue by 0xfeeddeadbeef** [dotnet/extensions#4967]
### Description

When a project that is using `Microsoft.Extensions.Http.Diagnostics` has a **direct dependency** on [version 3.0.0 of Microsoft.IO.RecyclableMemoryStream](https://www.nuget.org/packages/Microsoft.IO.RecyclableMemoryStream/3.0.0) package and you try to use `AddExtendedHttpClientLogging` with body content logging enabled, it will fail with `MissingMethodException: RecyclableMemoryStreamManager.GetStream()`.

### Reproduction Steps

1. Create a new ASP.NET Web API project
2. Add `<PackageReference Include="Microsoft.Extensions.Http.Diagnostics" Version="8.2.0" />`
3. Add `<PackageReference Include="Microsoft.IO.RecyclableMemoryStream" Version="3.0.0" />`
4. Add a HTTP client with body logging enabled: `clientBuilder.AddExtendedHttpClientLogging(x => { x.LogBody = true; });`
5. Send a request using this HTTP client

### Expected behavior

HTTP request and response bodies are logged without any errors.

### Actual behavior

```
An error occurred while reading the response data to fill the logger context for request: {http.request.method} {server.address}/{url.path} <s:Microsoft.Extensions.Http.Logging.HttpClientLogger>
System.MissingMethodException: Method not found: 'System.IO.MemoryStream Microsoft.IO.RecyclableMemoryStreamManager.GetStream()'.
   at Microsoft.Extensions.Http.Logging.Internal.HttpResponseBodyReader.ReadFromStreamAsync(HttpResponseMessage response, Int32 readSizeLimit, RecyclableMemoryStreamManager streamManager, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Extensions.Http.Logging.Internal.HttpResponseBodyReader.ReadFromStreamAsync(HttpResponseMessage response, Int32 readSizeLimit, RecyclableMemoryStreamManager streamManager, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.Internal.HttpResponseBodyReader.ReadFromStreamWithTimeoutAsync(HttpResponseMessage response, TimeSpan readTimeout, Int32 readSizeLimit, RecyclableMemoryStreamManager streamManager, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.Internal.HttpRequestReader.ReadResponseAsync(LogRecord logRecord, HttpResponseMessage response, List`1 responseHeadersBuffer, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.HttpClientLogger.LogResponseAsync(Object context, HttpRequestMessage request, HttpResponseMessage response, Exception exception, TimeSpan elapsed, CancellationToken cancellationToken)
```

### Known Workarounds

Use 2.x versions of Microsoft.IO.RecyclableMemoryStream package:

```
<PackageReference Include="Microsoft.IO.RecyclableMemoryStream" Version="2.3.2" />
```

**Comment by RussKie** [comment]
@dotnet/dotnet-r9-telemetry could you please investigate?

**Comment by xakep139** [comment]
Thanks for reporting this, @0xfeeddeadbeef 
It seems that version `3.0.0` of `Microsoft.IO.RecyclableMemoryStream` package has a bunch of breaking changes: https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream/releases/tag/3.0.0, including `GetStream()` method overloads.
Even though we can update our dependencies to rely on version 3 instead of 2, this might break existing users who rely on version 2.x in their code.
@geeknoid @joperezr thoughts?

**Comment by geeknoid** [comment]
I think we should take the leap and upgrade our dependency. It's the only way stuff will keep working in the long run, otherwise we'll fall behind forever.

**Comment by xakep139** [comment]
@0xfeeddeadbeef the changes will be released in version 8.3.0
